paymentintent.go

```diff
@@ -0,0 +1,134 @@
+
+import (
+	"encoding/json"
+)
+
+type PaymentIntentCaptureMethod string
+
+const (
+	PaymentIntentCaptureMethodAutomatic PaymentIntentCaptureMethod = "automatic"
+	PaymentIntentCaptureMethodManual    PaymentIntentCaptureMethod = "manual"
+)
+
+type PaymentIntentConfirmationMethod string
+
+const (
+	PaymentIntentConfirmationMethodPublishable PaymentIntentConfirmationMethod = "publishable"
+	PaymentIntentConfirmationMethodSecret      PaymentIntentConfirmationMethod = "secret"
+)
+
+type PaymentIntentStatus string
+
+const (
+	PaymentIntentStatusCanceled             PaymentIntentStatus = "canceled"
+	PaymentIntentStatusProcessing           PaymentIntentStatus = "processing"
+	PaymentIntentStatusRequiresCapture      PaymentIntentStatus = "requires_capture"
+	PaymentIntentStatusRequiresConfirmation PaymentIntentStatus = "requires_confirmation"
+	PaymentIntentStatusRequiresSource       PaymentIntentStatus = "requires_source"
+	PaymentIntentStatusRequiresSourceAction PaymentIntentStatus = "requires_source_action"
+	PaymentIntentStatusSucceeded            PaymentIntentStatus = "succeeded"
+)
+
+type PaymentIntentTransferDataParams struct {
+	Amount *int64 `form:"amount"`
+}
+
+// PaymentIntentParams is the set of parameters that can be used when handling a payment intent.
+type PaymentIntentParams struct {
+	Params               `form:"*"`
+	AllowedSourceTypes   []*string                        `form:"allowed_source_types"`
+	Amount               *int64                           `form:"amount"`
+	ApplicationFee       *int64                           `form:"application_fee"`
+	AttemptConfirmation  *string                          `form:"attempt_confirmation"`
+	CaptureMethod        *string                          `form:"capture_method"`
+	Currency             *string                          `form:"currency"`
+	Customer             *string                          `form:"customer"`
+	Description          *string                          `form:"description"`
+	OnBehalfOf           *string                          `form:"on_behalf_of"`
+	ReceiptEmail         *string                          `form:"receipt_email"`
+	ReturnURL            *string                          `form:"return_url"`
+	SaveSourceToCustomer *bool                            `form:"save_source_to_customer"`
+	Shipping             *ShippingDetailsParams           `form:"shipping"`
+	Source               *string                          `form:"source"`
+	StatementDescriptor  *string                          `form:"statement_descriptor"`
+	TransferData         *PaymentIntentTransferDataParams `form:"transfer_data"`
+	TransferGroup        *string                          `form:"transfer_group"`
+}
+
+// PaymentIntentListParams is the set of parameters that can be used when listing payment intents.
+// For more details see https://stripe.com/docs/api#list_payouts.
+type PaymentIntentListParams struct {
+	ListParams `form:"*"`
+}
+
+type PaymentIntentNextSourceActionValue struct {
```

Comment:
We will need to do something smarter here as the shape of `value` changes based on `type`. This will be closer to what we do with BalanceTransaction and `source`